src/net.cpp

```diff
@@ -2853,7 +2855,16 @@ int64_t CConnman::PoissonNextSendInbound(int64_t now, int average_interval_secon
 
 int64_t PoissonNextSend(int64_t now, int average_interval_seconds)
 {
-    return now + (int64_t)(log1p(GetRand(1ULL << 48) * -0.0000000000000035527136788 /* -1/2^48 */) * average_interval_seconds * -1000000.0 + 0.5);
+    const int64_t average_interval_microseconds = average_interval_seconds * 1000000ULL;
+    int64_t delta_microseconds = -std::log1p(GetRand(1ULL << 48) * -0.0000000000000035527136788 /* -1/2^48 */) * average_interval_microseconds + 0.5;
+
+    // Upper bound to avoid timeouts in the functional tests.
+    static const bool regtest = gArgs.GetChainName() == CBaseChainParams::REGTEST;
```

Comment:
I think this is really ugly, both doing a string compare here, making the function depend on a global, and doing this at first call (which is not *guaranteed* to be after initializing the chain name).
Though I don't really have a proposal how to do this differently.